nuxeo-common/src/main/java/org/nuxeo/common/utils/RFC2231.java

```diff
@@ -59,6 +66,38 @@ public static void percentEscape(StringBuilder sb, String value) {
         }
     }
 
+    /**
+     * Encodes a MIME parameter per RFC 2231.
+     * <p>
+     * This implementation always uses UTF-8 and no language.
+     *
+     * @param sb the buffer to fill
+     * @param value the value to encode
+     */
+    protected static void encodeRFC2231(StringBuilder sb, String value) {
+        int originalLength = sb.length();
+        sb.append("*=UTF-8''"); // no language
+        byte[] bytes = value.getBytes(UTF_8);
+        boolean encoded = false;
+        for (int i = 0; i < bytes.length; i++) {
+            int c = bytes[i] & 0xff;
+            if (c <= 32 || c >= 127 || RFC2231_SPECIALS.indexOf(c) != -1) {
+                sb.append('%');
+                sb.append(HEX_DIGITS[c >> 4]);
+                sb.append(HEX_DIGITS[c & 0xf]);
+                encoded = true;
+            } else {
+                sb.append((char) c);
+            }
+        }
+        if (!encoded) {
+            // undo and use basic format
+            sb.setLength(originalLength);
+            sb.append('=');
+            sb.append(value);
+        }
+    }
```

Comment:
Wouldn't it be clearer to use another StringBuilder `valueSb` for the possibly encoded value, then have something simple afterwards, like:
```
if (encoded) {
    sb.append("*=UTF-8''"); // no language
    sb.append(valueSb.toString());
} else {
    sb.append('=');
    sb.append(value);
}
```

Reply:
That was my initial code but it's an anti-pattern to have two StringBuilders and add one to the other. The `setLength` trick allows an easy "reset" to avoid allocating a new StringBuilder.

Reply:
It may be an anti-pattern but I still find it more explicit when reading the code.

Reply:
I think it's a matter of habit... It's useful to be able to "go back" in a StringBuilder.